clarity/contracts/stake-pool-diko.clar

```
@@ -190,11 +188,44 @@
   )
 )
 
+;; Sender can unstake all tokens without claiming rewards
+(define-public (emergency-withdraw)
+  (begin
+    ;; Save currrent cumm reward per stake
+    (increase-cumm-reward-per-stake)
+
+    (let (
+      ;; Calculate new stake amount
+      (amount (unwrap! (get-balance-of tx-sender) ERR-NOT-AUTHORIZED))
```

General comment here since I have a few of these as well in the code that I wrote, but

we probably need a generic error (like you have a status 500 on the web). Not authorized is not really applicable here. Everyone is authorized to see any principal's balance, but for whatever reason that goes wrong, we could throw an error 500-like error code.

An alternative would be to use `unwrap-panic` instead of `unwrap!`, but this could give issues on the front-end, since it "crashes" the whole smart contract call and doesn't allow for clean error catching in the front-end.

---

Would be better to catch the actual error but if that does not work with the FE, we need an alternative. Will change to 500.

---

So the actual error works. We can make a real error for it (e.g. not a generic 500), but what I'm saying is that `unwrap-panic` does not work since it throws some uncaught exception.